src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -131,57 +135,51 @@ public IReadOnlyNetworkPeerCollection ConnectedNodes
             IDateTimeProvider dateTimeProvider,
             INetworkPeerFactory networkPeerFactory)
         {
+            this.asyncLoopFactory = asyncLoopFactory;
+            this.connectedNodes = new NetworkPeerCollection();
             this.dateTimeProvider = dateTimeProvider;
+            this.loggerFactory = loggerFactory;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.Network = network;
+            this.networkPeerFactory = networkPeerFactory;
             this.NodeSettings = nodeSettings;
             this.nodeLifetime = nodeLifetime;
+            this.peerAddressManager = peerAddressManager;
+            this.Servers = new List<NetworkPeerServer>();
+
             this.Parameters = parameters;
             this.Parameters.ConnectCancellation = this.nodeLifetime.ApplicationStopping;
+            this.Parameters.UserAgent = $"{this.NodeSettings.Agent}:{this.GetVersion()}";
+            this.Parameters.Version = this.NodeSettings.ProtocolVersion;
 
-            this.loggerFactory = loggerFactory;
-            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
-
-            this.asyncLoopFactory = asyncLoopFactory;
+            // Don't start the peer discovery connector if the -connect arg is specified at startup.
+            if (!this.NodeSettings.ConnectionManager.Connect.Any())
+                this.DiscoverNodesPeerConnector = new PeerConnectorDiscovery(this.asyncLoopFactory, this.logger, this.Network, this.networkPeerFactory, this.nodeLifetime, this.NodeSettings, this.peerAddressManager);
 
-            this.peerAddressManager = peerAddressManager;
-            this.networkPeerFactory = networkPeerFactory;
+            // Use if we have specified the -connect arg at startup.
+            if (this.NodeSettings.ConnectionManager.Connect.Any())
+                this.ConnectNodePeerConnector = new PeerConnectorConnectNode(this.asyncLoopFactory, this.logger, this.Network, this.networkPeerFactory, this.nodeLifetime, this.NodeSettings, this.peerAddressManager);
 
-            this.Servers = new List<NetworkPeerServer>();
-            this.connectedNodes = new NetworkPeerCollection();
+            // Use if we have specified the -addnode arg at startup.
+            this.AddNodePeerConnector = new PeerConnectorAddNode(this.asyncLoopFactory, this.logger, this.Network, this.networkPeerFactory, this.nodeLifetime, this.NodeSettings, this.peerAddressManager);
         }
 
         public void Start()
         {
             this.logger.LogTrace("()");
 
-            this.Parameters.UserAgent = $"{this.NodeSettings.Agent}:{this.GetVersion()}";
-            this.Parameters.Version = this.NodeSettings.ProtocolVersion;
-
+            // TODO: Move connection manager to the connector next
```

Comment:
Yes good idea also it seems `ConnectionManager` can be injected in to Connector.

Reply:
Will do in next PR 👍